File: App/src/Adaptive.ReactiveTrader.Client.Domain/Models/Pricing/PriceFactory.cs
using Adaptive.ReactiveTrader.Client.Domain.Models.ReferenceData;
using Adaptive.ReactiveTrader.Client.Domain.Repositories;
using Adaptive.ReactiveTrader.Shared.DTO.Pricing;

namespace Adaptive.ReactiveTrader.Client.Domain.Models.Pricing
{
    internal class PriceFactory : IPriceFactory
    {
        private readonly IExecutionRepository _executionRepository;

        public PriceFactory(IExecutionRepository executionRepository)
        {
            _executionRepository = executionRepository;
        }

        public IPrice Create(PriceDto priceDto, ICurrencyPair currencyPair)
        {
            var bid = new ExecutablePrice(Direction.SELL, priceDto.Bid, _executionRepository);
            var ask = new ExecutablePrice(Direction.BUY, priceDto.Ask, _executionRepository);
            var price = new Price(bid, ask, priceDto.Mid, priceDto.QuoteId, priceDto.ValueDate, currencyPair);

            return price;
        }
    }
}

File: App/src/Adaptive.ReactiveTrader.Client.JS/Models/Pricing/PriceFactory.ts
class PriceFactory implements IPriceFactory {
    private _executionRepository: IExecutionRepository;

    constructor(executionRepository: IExecutionRepository) {
        this._executionRepository = executionRepository;
    }

    create(priceDto: PriceDto, currencyPair: ICurrencyPair) {
        var bid = new ExecutablePrice(Direction.Sell, priceDto.Bid, this._executionRepository);
        var ask = new ExecutablePrice(Direction.Buy, priceDto.Ask, this._executionRepository);
        var price = new Price(bid, ask, priceDto.Mid, priceDto.QuoteId, priceDto.ValueDate, currencyPair);

        return price;
    }
} 

